app/models/manageiq/providers/azure/cloud_manager/provision_workflow.rb

```diff
@@ -15,6 +15,19 @@ def allowed_resource_groups(_options = {})
     resource_groups.each_with_object({}) { |rg, hash| hash[rg.id] = rg.name }
   end
 
+  def allowed_cloud_networks(_options = {})
+    source = load_ar_obj(get_source_vm)
+    ems = source.try(:ext_management_system)
+
+    if ems.present?
+      ems.cloud_networks.select{ |cn| cn.cloud_subnets.size > 0 }.each_with_object({}) do |cn, hash|
```

Comment:
This is an N+1...can you use one of the suggestions we gave in gitter?

Reply:
Updated.